mmv1/third_party/terraform/.teamcity/PERFORMING_TASKS_IN_TEAMCITY.md

```diff
@@ -39,7 +39,9 @@ Google Cloud/
 
 ### Looking at nightly test results
 
-A CRON trigger causes all acceptance tests for the GA and Beta providers to run overnight, Mon-Sun. These tests use the `main` branches of hashicorp/terraform-provider-google(-beta) repos. The tests interact fully with Google APIs and are used to identify any breaking changes introduced by either a recent PR or a change in the API itself.
+A CRON trigger causes all acceptance tests for the GA and Beta providers to run overnight, Mon-Sun. These tests use the `nightly-test` branches of hashicorp/terraform-provider-google(-beta) repos that are created by checking out of `main` every night through the [`TeamCity - nightly` Github Action](https://github.com/hashicorp/terraform-provider-google/actions/workflows/teamcity-nightly-workflow.yaml). (A separate branch is necessary in order to ensure that all tests are ran on the same commit. Services are separated in their own builds and are sent into an agent queue. Depending on the timing of the queue a build could potentially have a different commit from the rest if `main` received a commit prior to exiting.) 
```

Comment:
This section could benefit from using some bullet points to include the reasoning for not using main, here's an illustration:

```
What is the `nightly-test` branch?:
* A branch created by the [X GHA](link) as a way to select a recent commit on main that all tests should run on
* The branch is recreated each night
* If we use the main branch directly instead: Services are separated in their own builds and are sent into an agent queue. Depending on the timing of the queue a build could potentially have a different commit from the rest if `main` received a commit prior to exiting.
```

This can help people scan the text for information a bit more easily (and potentially skip over bits they're not interested in)